builtin/logical/aws/path_static_roles.go

```diff
@@ -0,0 +1,300 @@
+					Description: descRotationPeriod,
+				},
+			},
+		}},
+	}
+
+	return &framework.Path{
+		Pattern: fmt.Sprintf("%s/%s", pathStaticRole, framework.GenericNameWithAtRegex(paramRoleName)),
+		Fields: map[string]*framework.FieldSchema{
+			paramRoleName: {
+				Type:        framework.TypeString,
+				Description: descRoleName,
+			},
+			paramUsername: {
+				Type:        framework.TypeString,
+				Description: descUsername,
+			},
+			paramRotationPeriod: {
+				Type:        framework.TypeDurationSecond,
+				Description: descRotationPeriod,
+			},
+		},
+
+		Operations: map[logical.Operation]framework.OperationHandler{
+			logical.ReadOperation: &framework.PathOperation{
+				Callback:  b.pathStaticRolesRead,
+				Responses: roleResponse,
+			},
+			logical.UpdateOperation: &framework.PathOperation{
+				Callback:                    b.pathStaticRolesWrite,
+				ForwardPerformanceSecondary: true,
+				ForwardPerformanceStandby:   true,
+				Responses:                   roleResponse,
+			},
+			logical.DeleteOperation: &framework.PathOperation{
+				Callback:                    b.pathStaticRolesDelete,
+				ForwardPerformanceSecondary: true,
+				ForwardPerformanceStandby:   true,
+				Responses: map[int][]framework.Response{
+					http.StatusNoContent: {{
+						Description: http.StatusText(http.StatusNoContent),
+					}},
+				},
+			},
+		},
+
+		HelpSynopsis:    pathStaticRolesHelpSyn,
+		HelpDescription: pathStaticRolesHelpDesc,
+	}
+}
+
+func (b *backend) pathStaticRolesRead(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	roleName, ok := data.GetOk(paramRoleName)
+	if !ok {
+		return nil, fmt.Errorf("missing '%s' parameter", paramRoleName)
+	}
+
+	b.roleMutex.RLock()
+	defer b.roleMutex.RUnlock()
+
+	entry, err := req.Storage.Get(ctx, formatRoleStoragePath(roleName.(string)))
+	if err != nil {
+		return nil, fmt.Errorf("failed to read configuration for static role %q: %w", roleName, err)
+	}
+	if entry == nil {
+		return nil, nil
+	}
+
+	var config staticRoleEntry
+	if err := entry.DecodeJSON(&config); err != nil {
+		return nil, fmt.Errorf("failed to decode configuration for static role %q: %w", roleName, err)
+	}
+
+	return &logical.Response{
+		Data: formatResponse(config),
+	}, nil
+}
+
+func (b *backend) pathStaticRolesWrite(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	// Create & validate config from request parameters
+	config := staticRoleEntry{}
+
+	if rawRoleName, ok := data.GetOk(paramRoleName); ok {
+		config.Name = rawRoleName.(string)
+
+		if err := b.validateRoleName(config.Name); err != nil {
+			return nil, err
+		}
+	} else {
+		return nil, fmt.Errorf("missing '%s' parameter", paramRoleName)
+	}
+
+	if rawUsername, ok := data.GetOk(paramUsername); ok {
+		config.Username = rawUsername.(string)
+
+		if err := b.validateIAMUserExists(ctx, req, config.Username); err != nil {
```

Comment:
yeah, it does look like we're upserting the config between validating the user and getting the creds, which raises the possibility that you could end up with an entry that has no creds and connects to nothing. I don't think that would be too different than any other time the user could have been deleted out from under us, but we could probably take more pains to be atomic.

This also reminds me I should ensure `DELETE` still works if the IAM user is gone, since we need to leave a way to clean up a role in that case.